pages/help/HelpHandler.inc.php

```diff
@@ -44,15 +50,27 @@ function index($args, $request) {
 		$language = AppLocale::getIso1FromLocale(AppLocale::getLocale());
 		if (!file_exists($path . $language)) $language = 'en'; // Default
 
-		if (!$filename || !preg_match('#^([[a-zA-Z0-9_-]+/)+[a-zA-Z0-9_-]+\.\w+$#', $filename) || !file_exists($path . $filename)) {
-			$request->redirect(null, null, null, array($language, 'SUMMARY.md'));
+		if (!$isPluginHelpRequest || !file_exists($this->getPathForPluginHelpChapter($request, $language))) {
+			if (!$filename || !preg_match('#^([[a-zA-Z0-9_-]+/)+[a-zA-Z0-9_-]+\.\w+$#', $filename) || !file_exists($path . $filename)) {
+				$request->redirect(null, null, null, array($language, 'SUMMARY.md'));
+			}
 		}
 
+		// Give plugins the opportunity to provide help chapters for the toc
+		$pluginHelpChapters = array();
+		HookRegistry::call('Plugins::Help', array(&$pluginHelpChapters));
```

Comment:
So, the format you expect is like this:

```
$pluginHelpChapters['Citation Style Language'] = '/path/to/help/file';
```

I think we'll run into issues trying to use titles in array keys, with invalid characters needing to be used. Let's use a structure with a `label` and `url` keys:

```
$pluginHelpChapters[] = array(
  'label' => __('translation.key'),
  'file' => 'index.md',
);
```

Reply:
Also, can we call this hook `Help::Plugins` since it's owned by the `HelpHandler`?